### ISSUE TITLE
[a11y] External unstyled link causes change of context without warning

### ISSUE THREAD

**Issue by mzlock** [Shopify/polaris#3476]
## Issue summary

Related to: https://github.com/Shopify/polaris-ux/issues/146

A solution to this issue was already found and deployed for the `<Link>` component (https://github.com/Shopify/polaris-react/pull/1247), but the problem persists for a `<UnstyledLink>` (https://github.com/Shopify/polaris-react/blob/master/src/components/UnstyledLink/UnstyledLink.tsx)

## Expected behavior

If a `<UnstyledLink>` is going to open in a new tab, then a screen reader should notify the user that the link will open in a new tab.

## Actual behavior

The user is not notified that the link will open in a new tab.

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 5.7.0

## Fix?

Since an Unstyled Link likely should not have [the external icon, like the Link has](https://github.com/Shopify/polaris-react/pull/1247/files#diff-54ce4d51798a73c9699af609ffeb83195ebdefc49ce0ebec4f44432d97c383b1R43), then it should have some `<VisuallyHidden>` text instead:

```
<VisuallyHidden>
  <span>{ polaris.intl.translate('Polaris.Common.newWindowAccessibilityHint') }</span>
</VisuallyHidden>
```

**Comment by BPScott** [comment]
See https://github.com/Shopify/polaris-react/pull/3529#issuecomment-714680749 for thoughts on issues with pushing the responsiblity of handling this text onto UnstyledLink / our consumers who define a custom LinkComponent
